src/grasp_scorer.cpp

```diff
@@ -250,4 +250,120 @@ Eigen::Vector3d GraspScorer::scoreRotationsFromDesired(const Eigen::Isometry3d&
   return scores;
 }
 
+double GraspScorer::scoreSuctionVoxelOverlap(const Eigen::Isometry3d& grasp_pose_tcp, const GraspDataPtr& grasp_data,
+                                             const Eigen::Isometry3d& object_pose, const Eigen::Vector3d& object_size,
+                                             std::vector<double> overlap_vector,
+                                             moveit_visual_tools::MoveItVisualToolsPtr visual_tools)
+{
+  overlap_vector.resize(grasp_data->suction_voxel_matrix_->getNumVoxels());
+
+  // These are in the object pose frame
+  Eigen::Vector3d box_bb_max(object_size.x() / 2.0, object_size.y() / 2.0, 0);
+  Eigen::Vector3d box_bb_min(-object_size.x() / 2.0, -object_size.y() / 2.0, 0);
+
+  if (visual_tools)
+  {
+    visual_tools->deleteAllMarkers();
+    visual_tools->trigger();
+    visual_tools->publishAxisLabeled(grasp_pose_tcp, "tcp");
+    Eigen::Isometry3d bbb1 =
+        object_pose * Eigen::Isometry3d(Eigen::Translation3d(box_bb_max) * Eigen::Quaterniond(1, 0, 0, 0));
+    Eigen::Isometry3d bbb2 =
+        object_pose * Eigen::Isometry3d(Eigen::Translation3d(box_bb_min) * Eigen::Quaterniond(1, 0, 0, 0));
+    visual_tools->publishAxisLabeled(bbb1, "bbb1", rviz_visual_tools::SMALL);
+    visual_tools->publishAxisLabeled(bbb2, "bbb2", rviz_visual_tools::SMALL);
+    visual_tools->trigger();
+  }
+
+  double voxel_area = grasp_data->suction_voxel_matrix_->getVoxelArea();
+  Eigen::Isometry3d grasp_pose_tcp_in_object_frame = object_pose.inverse() * grasp_pose_tcp;
+  for (std::size_t voxel_id = 0; voxel_id < grasp_data->suction_voxel_matrix_->getNumVoxels(); ++voxel_id)
+  {
+    std::shared_ptr<SuctionVoxel> voxel;
+    if (!grasp_data->suction_voxel_matrix_->getSuctionVoxel(voxel_id, voxel))
+    {
+      ROS_ERROR_STREAM_NAMED("grasp_scorer.voxels", "voxel id: " << voxel_id << " is out of range");
+      overlap_vector.clear();
+      overlap_vector.resize(grasp_data->suction_voxel_matrix_->getNumVoxels());
+      return 0;
+    }
+
+    // These are also in the object pose frame
+    Eigen::Vector3d tmp_voxel_bb_1 = grasp_pose_tcp_in_object_frame * voxel->bottom_left_;
+    Eigen::Vector3d tmp_voxel_bb_2 = grasp_pose_tcp_in_object_frame * voxel->top_right_;
+    Eigen::Vector3d voxel_bb_min = Eigen::Vector3d(std::min(tmp_voxel_bb_1.x(), tmp_voxel_bb_2.x()),
```

Comment:
yup. I was avoiding that use case since almost all use cases involve grasp candidates that are axis aligned. I went through and implemented a summation of slices approach that works well.